zkevm-circuits/src/rlp_circuit_fsm.rs

```diff
@@ -2256,6 +2226,227 @@ impl<F: Field> RlpCircuitConfig<F> {
+                        - meta.query_advice(depth, Rotation::cur())
+                    ) * (
+                        meta.query_advice(byte_rev_idx, Rotation::cur()) - 1.expr()
+                    ),
+                );
+            });
+
+            cb.gate(and::expr([
+                meta.query_fixed(q_enabled, Rotation::cur()),
+                not::expr(is_end(meta)),
+            ]))
+        });
+
+        meta.create_gate("Decoding table stack op INIT correspondence", |meta| {
+            let mut cb = BaseConstraintBuilder::default();
+
+            cb.require_equal(
+                "stack init pushes all remaining_bytes onto depth 0",
+                meta.query_advice(rlp_decoding_table.value, Rotation::cur()),
+                meta.query_advice(byte_rev_idx, Rotation::cur()),
+            );
+            cb.require_equal(
+                "stack can only init once with the first byte",
+                meta.query_advice(byte_idx, Rotation::cur()),
+                1.expr(),
+            );
+            cb.require_zero(
+                "stack inits at depth 0",
+                meta.query_advice(rlp_decoding_table.depth, Rotation::cur()),
+            );
+            cb.require_zero(
+                "stack inits with al_idx at 0",
+                meta.query_advice(rlp_decoding_table.al_idx, Rotation::cur()),
+            );
+            cb.require_zero(
+                "stack inits with sk_idx at 0",
+                meta.query_advice(rlp_decoding_table.sk_idx, Rotation::cur()),
+            );
+            cb.condition(
+                not::expr(meta.query_fixed(q_first, Rotation::cur())),
+                |cb| {
+                    cb.require_boolean(
+                        "tx_id can only stay the same or increment by 1",
+                        meta.query_advice(rlp_decoding_table.tx_id, Rotation::cur())
+                            - meta.query_advice(rlp_decoding_table.tx_id, Rotation::prev()),
+                    );
+                },
+            );
+            cb.require_equal(
+                "Init Op has DecodeTagStart in the state machine",
+                is_decode_tag_start(meta),
+                1.expr(),
+            );
+
+            cb.gate(and::expr([
+                meta.query_fixed(q_enabled, Rotation::cur()),
+                meta.query_advice(rlp_decoding_table.is_stack_init, Rotation::cur()),
+            ]))
+        });
+
+        meta.lookup_any("Decoding table stack op PUSH correspondence", |meta| {
+            let enable = meta.query_advice(is_push_op_lookup, Rotation::cur());
+
+            let input_exprs = vec![
+                meta.query_advice(tx_id, Rotation::cur()),
+                meta.query_advice(format, Rotation::cur()),
+                meta.query_advice(byte_idx, Rotation::cur()),
+                meta.query_advice(depth, Rotation::cur()) + 1.expr(),
+                1.expr(),
+                select::expr(
```

Comment:
Should you add `al_idx` in the input_exprs?

Reply:
`al_idx` is now part of input_exprs.